packages/fast-usdc-contract/src/fast-usdc.flows.ts

```diff
@@ -1,4 +1,30 @@
-import type { OrchestrationFlow, Orchestrator } from '@agoric/orchestration';
+import type { NatAmount } from '@agoric/ertp';
+import type { EvmHash } from '@agoric/fast-usdc/src/types.ts';
+import type {
+  AccountId,
+  OrchestrationAccount,
+  OrchestrationFlow,
+  Orchestrator,
+} from '@agoric/orchestration';
+import { parseAccountId } from '@agoric/orchestration/src/utils/address.js';
+import { assertAllDefined, makeTracer } from '@agoric/internal';
+import type { StatusManager } from './exos/status-manager.ts';
+
+const FORWARD_TIMEOUT = {
+  sec: 10n * 60n,
+  p: '10m',
+} as const;
+harden(FORWARD_TIMEOUT);
+
+export interface Context {
+  /** e.g., `agoric-3` */
+  currentChainReference: string;
+  supportsCctp: (destination: AccountId) => boolean;
+  log?: Console['log'];
+  statusManager: StatusManager;
+  getNobleICA: () => OrchestrationAccount<{ chainId: 'noble-1' }>;
+  settlementAccount: Promise<OrchestrationAccount<{ chainId: 'agoric-any' }>>;
```

Comment:
thinking out loud... we pass a callback to get the noble ICA, but a promise to get the settlementAccount. seems a little inconsistent. I suppose there are constraints that motivate the different approaches.

Reply:
Yeah, I share the concern. What I could do is refactor it also to a promise but I'm not sure it's better,
```js
   const nobleIcaPK = makePromiseKit();
   if (baggage.has(NOBLE_ICA_BAGGAGE_KEY))
       nobleIcaPK.resolve(baggage.get(NOBLE_ICA_BAGGAGE_KEY));
```

And then whenever it is set, also resolve it.

Maybe there's some "durably stored promise" kit abstraction we could make, but still I don't think it's worth it.

What I will do is at a comment justifying the disharmony.

Reply:
I meant a regular promise that gets durably stored, like is happening here. "durably stored promise result kit" may have been clearer.

Reply:
native promises cannot be durably stored. The design for durable promises puts them more in line with vows than promises.

I think I still don't understand what "durably stored promise result kit" means.